src/d3a/d3a_core/simulation.py

```diff
@@ -356,13 +358,18 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
         if self.export_on_finish:
             log.info("Exporting simulation data.")
             if GlobalConfig.POWER_FLOW:
-                self.export.export(power_flow=self.power_flow)
+                self.export.export(export_plots=self.should_export_plots,
+                                   power_flow=self.power_flow)
             else:
-                self.export.export()
+                self.export.export(self.should_export_plots)
 
         if self.use_repl:
             self._start_repl()
 
+    @property
+    def should_export_plots(self):
+        return not self.redis_connection.is_enabled()
+
```

Comment:
In line 338 the CLI flag is also incorporated: 
`if self.export_on_finish or self.redis_connection.is_enabled():`
How about adding it here too?

Reply:
`self.export_on_finish` is responsible for controlling whether the csv files are exported too. In the case of redis connection, we still want the csv files to be produced, we do not want the plots to be exported though.

Reply:
Yes, you are right. Ignore my comment then.